holoviews/operation/datashader.py

```diff
@@ -145,12 +153,19 @@ def get_agg_data(cls, obj, category=None):
         elif isinstance(obj, Element):
             glyph = 'line' if isinstance(obj, Curve) else 'points'
             paths.append(obj.data if is_df(obj) else obj.dframe())
-        if glyph == 'line':
-            empty = paths[0][:1].copy()
-            empty.loc[0, :] = (np.NaN,) * empty.shape[1]
-            paths = [elem for path in paths for elem in (path, empty)][:-1]
         if len(paths) > 1:
-            df = pd.concat(paths).reset_index(drop=True)
+            if glyph == 'line':
+                path = paths[0][:1]
+                if isinstance(path, dd.DataFrame):
+                    path = path.compute()
+                empty = path.copy()
+                empty.loc[0, :] = (np.NaN,) * empty.shape[1]
+                paths = [elem for path in paths for elem in (path, empty)][:-1]
+            datasets = [Dataset(p) for p in paths]
+            if isinstance(paths[0], dd.DataFrame):
```

Comment:
``isinstance`` checks over data formats is just the sort of thing interfaces are supposed to handle for you. I am hoping we can get rid of these isinstance checks, perhaps by using the appropriate utility to select the right interface based on the data type (i.e whatever dataframe type it happens to be)?